app/service/authenticated_service_info.py

```diff
@@ -0,0 +1,183 @@
+        """
+        self.created_at = key.created_at
+        self.created_by = key.created_by
+        self.created_by_id = key.created_by_id
+        self.expiry_date = key.expiry_date
+        self.id = key.id
+        self.key_type = key.key_type
+        self.name = key.name
+        self.secret = key.secret
+
+        _service = {
+            "research_mode": key.service.research_mode,
+            "restricted": key.service.restricted,
+            "active": key.service.active,
+        }
+        self.service = SimpleNamespace(**_service)
+        self.service_id = key.service_id
+        self.updated_at = key.updated_at
+
+
+class AuthenticatedServiceInfo:
+    """
+    Represents the relevant information extracted from an SQLAlchemy query result.
+
+    This class was created to allow for reading from multiple databases and to facilitate caching.
+    By extracting only the necessary information from the query result, it ensures compatibility
+    across different database connections and enables efficient serialization for caching purposes.
+    The extraction of specific information rather than working with entire query results provides a
+    more controlled and optimized way to handle the data.
+
+    Attributes:
+        active (bool): A flag indicating whether the service is active.
+        permissions (list): A list of permissions associated with the service.
+        api_keys (list): A list of API keys associated with the service.
+        id (int): The unique identifier of the service.
+        research_mode (bool): A flag indicating whether the service is in research mode.
+        restricted (bool): A flag indicating whether the service is restricted.
+        rate_limit (int): The rate limit applied to the service.
+        service_sms_senders (list): A list of SMS senders associated with the service.
+        message_limit (int): The message limit applied to the service.
+        users (list): A list of users associated with the service.
+        whitelist (list): A list of whitelisted entities for the service.
+
+    Methods:
+        <List the methods here, if applicable>
+    """
+
+    def __init__(self, result=None):
+        self.active = None
+        self.permissions = None
+        self.api_keys = None
+        self.id = None
+        self.research_mode = None
+        self.restricted = None
+        self.rate_limit = None
+        self.service_sms_senders = None
+        self.message_limit = None
+        self.users = None
+        self.whitelist = None
+        if result is not None:
+            try:
+                self.extract(result)
+            except Exception as err:
+                raise AuthenticatedServiceInfoException(err)
+
+    def extract(self, result):
+        """
+        Extracts the necessary data from an authenticated service ORM object.
+
+        Args:
+            result: The ORM object containing the data.
+
+        Returns:
+            None
```

Comment:
I appreciate that you included this, but please use an annotation, like you did on line 53.
```python
def extract(self,result) -> None:
```